tests/test-distribution/src/main/java/org/eclipse/jetty/tests/distribution/DistributionTester.java

```diff
@@ -0,0 +1,722 @@
+                    String[] match = new String[count];
+                    while (num++ < count) {
+                        match[num - 1] = mat.group(num);
+                    }
+                    pat.getMatches().add(match);
+
+                    if (pat.getCount() > 0) {
+                        getLatch().countDown();
+                    }
+                }
+            }
+        }
+
+        public void waitForDone(long timeout, TimeUnit unit)
+                throws InterruptedException {
+            getLatch().await(timeout, unit);
+        }
+
+        private CountDownLatch getLatch() {
+            synchronized (this) {
+                if (latch == null) {
+                    latch = new CountDownLatch(count);
+                }
+            }
+            return latch;
+        }
+    }
+
+    private static class ConsolePattern {
+        private Pattern pattern;
+
+        private List<String[]> matches;
+
+        private int count;
+
+        ConsolePattern(String exp, int cnt) {
+            pattern = Pattern.compile(exp);
+            matches = new ArrayList<>();
+            count = cnt;
+        }
+
+        public Matcher getMatcher(String line) {
+            return pattern.matcher(line);
+        }
+
+        public List<String[]> getMatches() {
+            return matches;
+        }
+
+        public int getCount() {
+            return count;
+        }
+    }
+
+
+    public void installWarFile(File warFile, String context) throws IOException {
+        //webapps
+        Path webapps = Paths.get(jettyBase.toString(),"webapps", context);
+        if (!Files.exists(webapps)) {
+            Files.createDirectories(webapps);
+        }
+        unzip(warFile, webapps.toFile());
+    }
+
+    //---------------------------------------
+    // Assert methods
+    //---------------------------------------
+
+    public void assertLogsContains(String txt){
+        assertTrue(logs.stream().filter(s -> StringUtils.contains(s, txt)).count()>0);
+    }
+
+
+    public void assertUrlStatus(String url, int expectedStatus){
+        try {
+            ContentResponse contentResponse = httpClient.GET(getBaseUri() + url);
+            int status = contentResponse.getStatus();
+            assertEquals(expectedStatus, status, () -> "status not " + expectedStatus + " but " + status);
+        } catch (InterruptedException|ExecutionException|TimeoutException e) {
+            fail(e.getMessage(),e);
+        }
+    }
+
+    public void assertUrlContains(String url, String content){
+        try {
+            ContentResponse contentResponse = httpClient.GET(getBaseUri() + url);
+            String contentResponseStr = contentResponse.getContentAsString();
+            assertTrue(StringUtils.contains(contentResponseStr, content), () -> "content not containing '" + content + "'");
+        } catch (InterruptedException|ExecutionException|TimeoutException e) {
+            fail(e.getMessage(),e);
+        }
+    }
```

Comment:
Why? This class is used only for testing a distribution so it's very convenient to have those methods. Otherwise we will have a lot duplicate code doing the same thing: create start http client send a request some asserts on the result then stop the client.
I definitely prefer not having to copy/paste this code in all the test we will write.

Reply:
Definitely `HttpClient` should be outside of `DistributionTester`.
The reason is A) to make one class do one thing only, and B) you have now an "hello-world" case where it _seems_ ok to have `HttpClient` inside `DistributionTester`, but the next thing I am going to do is to start the distro with http2 and make a HTTP/2 call, which now requires the test to control `HttpClient`'s transport.
Then I want to do the same with Unix sockets, and so on.

`HttpClient` does not belong to `DistributionTester`. Remove it.

I don't care about 2 lines of duplications in tests; I want to read a test case and know everything, not having to lookup the implementation to answer "what transport is the internal `HttpClient` configured with?".